src/functions/onDisconnect.ts

```diff
@@ -15,13 +48,29 @@ export const handler: Handler = async (event: WebSocketAPIGatewayEvent): Promise
   const { connectionId } = event.requestContext;
 
   console.log('Deleting connectionId from the db table...');
+  const ws = new WebSocketClient(event.requestContext);
   try {
     // Delete user from ConnectionsTable
-    await deleteConnection(connectionId);
+    const deletedUser = await deleteConnection(connectionId);
+
+    // Additional cleanup when user disconnect
+    if (deletedUser) {
+      // 1. Remove the disconnected user from the game
+      const updatedGame = await cleanupUserWhenDisconnect(ws, deletedUser);
+
+      // 2. Delete the game if the user is the host of the game and send updates to other users
+      // (see comments in the function for more details)
+      if (updatedGame) await sendUpdates(ws, deletedUser.connectionId, updatedGame);
+    }
+
+    // Send USER_UPDATE with DISCONNECT to all other users
+    const connectionIds = getConnectionIdsFromUsers(await getAllConnections());
+    const connectionIdsExceptCaller = connectionIds.filter((otherConnectionId) => otherConnectionId !== connectionId);
+    await broadcastUserUpdate(ws, connectionId, UserStates.DISCONNECTED, connectionIdsExceptCaller);
```

Comment:
in broadcastUserUpdate, currentUser is undefined as it is no longer in ConnectionsTable, and the message does not get sent

Reply:
ok, will fix it in next commit